components/contract-components/contract-deploy-form/built-in-contract.tsx

```diff
@@ -832,6 +836,8 @@ const BuiltinContractForm: React.FC<BuiltinContractFormProps> = ({
                   parseInt(e.currentTarget.value) as SUPPORTED_CHAIN_ID,
                 )
               }
+              disabledChainIds={DisabledChainsMap[contractType as ContractType]}
```

Comment:
since there can now be disabled chains we have to not allow the `value` of this to be set to one of the `disabledChainIds`

might make sense to do it within the `<SupportedNetworkSelect>`

Reply:
(basically the problem is currently I can connect my wallet to mainnet and it will show the <Deploy>` button as enabled since I didn't update it